exporter/exporterhelper/batch_sender.go

```diff
@@ -176,35 +183,34 @@ func (bs *batchSender) sendMergeSplitBatch(ctx context.Context, req Request) err
 		}
 		bs.mu.Unlock()
 		<-batch.done
-		if batch.err != nil {
-			return batch.err
-		}
+		errs = multierr.Append(errs, batch.err)
 		reqs = reqs[1:]
 	} else {
 		bs.mu.Unlock()
 	}
 
-	// Intentionally do not put the last request in the active batch to not block it.
-	// TODO: Consider including the partial request in the error to avoid double publishing.
-	for _, r := range reqs {
-		if err := bs.nextSender.send(ctx, r); err != nil {
-			return err
-		}
+	if len(reqs) > 0 {
+		sendErr := bs.nextSender.send(ctx, reqs...)
+		errs = multierr.Append(errs, sendErr)
 	}
-	return nil
+
+	return errs
 }
 
 // sendMergeBatch sends the request to the batch and waits for the batch to be exported.
-func (bs *batchSender) sendMergeBatch(ctx context.Context, req Request) error {
+func (bs *batchSender) sendMergeBatch(ctx context.Context, reqs ...Request) error {
 	bs.mu.Lock()
 
+	var req Request
 	if bs.activeBatch.request != nil {
 		var err error
-		req, err = bs.mergeFunc(ctx, bs.activeBatch.request, req)
+		req, err = bs.mergeFunc(ctx, bs.activeBatch.request, reqs[0])
```

Comment:
Why do we ignore other requests?

Reply:
The input slice of Requests will have length 1 otherwise we need to use sendMergeSplitBatch, so it makes more sense to change this function signature back to `sendMergeBatch(ctx context.Context, req Request) error` and just pass the single request directly